core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -288,13 +292,21 @@ class UnifiedLog(@volatile var logStartOffset: Long,
 
   @volatile var partitionMetadataFile : PartitionMetadataFile = null
 
+  //todo-tier it needs to be updated.
+  private val localLogStartOffset: Long = logStartOffset
```

Comment:
It is a `val` for now and it is not needed to be declared as volatile and the compiler does not allow too. This will be updated in the upcoming changes to `var` and then it can be declared as volatile.